test/VectorExt/tiled_loop_masking.mlir

```diff
@@ -0,0 +1,52 @@
+// RUN: mlir-proto-opt %s -test-vector-masking-utils -split-input-file | FileCheck %s
+
+func @tiled_loop_1d(%arg0: tensor<?xf32>,
+                    %arg1: tensor<?xf32>) -> tensor<?xf32> {
+  %c0 = arith.constant 0 : index
+  %c8 = arith.constant 8 : index
+  %0 = tensor.dim %arg0, %c0 : tensor<?xf32>
+  %1 = linalg.init_tensor [%0] : tensor<?xf32>
+  %3 = linalg.tiled_loop (%arg2) = (%c0) to (%0) step (%c8) ins (%arg3 = %arg0: tensor<?xf32>, %arg4 = %arg1: tensor<?xf32>) outs (%arg5 = %1: tensor<?xf32>) {
+    %4 = affine.min affine_map<(d0)[s0] -> (8, -d0 + s0)>(%arg2)[%0]
+    %5 = tensor.extract_slice %arg3[%arg2] [%4] [1] : tensor<?xf32> to tensor<?xf32>
+    %6 = affine.min affine_map<(d0)[s0] -> (8, -d0 + s0)>(%arg2)[%0]
+    %7 = tensor.extract_slice %arg4[%arg2] [%6] [1] : tensor<?xf32> to tensor<?xf32>
+    %8 = affine.min affine_map<(d0)[s0] -> (8, -d0 + s0)>(%arg2)[%0]
+    %9 = tensor.extract_slice %arg5[%arg2] [%8] [1] : tensor<?xf32> to tensor<?xf32>
+    %10 = linalg.generic {indexing_maps = [affine_map<(d0) -> (d0)>, affine_map<(d0) -> (d0)>, affine_map<(d0) -> (d0)>],
+                          iterator_types = ["parallel"]}
+                          ins(%5, %7 : tensor<?xf32>, tensor<?xf32>) outs(%9 : tensor<?xf32>) {
+    ^bb0(%arg6: f32, %arg7: f32, %arg8: f32):  // no predecessors
+      %12 = arith.addf %arg6, %arg7 : f32
+      linalg.yield %12 : f32
+    } -> tensor<?xf32>
+    %11 = tensor.insert_slice %10 into %arg5[%arg2] [%8] [1] : tensor<?xf32> into tensor<?xf32>
+    linalg.yield %11 : tensor<?xf32>
+  }
+  return %3 : tensor<?xf32>
+}
+
+// CHECK-LABEL:   func @tiled_loop_1d(
+// CHECK:           %[[ZERO:.*]] = arith.constant 0 : index
+// CHECK:           %[[STEP:.*]] = arith.constant 8 : index
+// CHECK:           %[[UB:.*]] = tensor.dim
+// CHECK:           linalg.tiled_loop (%[[IV:.*]]) = (%[[ZERO]]) to (%[[UB]]) step (%[[STEP]])
+// CHECK-NEXT:        %[[PRED_MIN:.*]] = affine.min #map0(%[[UB]], %[[IV]])
+// CHECK-NEXT:        %[[PRED:.*]] = vector.create_mask %[[PRED_MIN]] : vector<8xi1>
+// CHECK-NEXT:        %[[PRED_RES:.*]] = vector_ext.predicate(%[[PRED]]: vector<8xi1>) -> (tensor<?xf32>) {
```

Comment:
Nice!

@ThomasRaoux here is some IR that would look similar to what we discussed yesterday for vector distribution.
Maybe the `vector.predicate` itself can be extended to also serve the GPU programming model purpose?

Reply:
It would be great to know more about the vector distribution! I think the vector.predicate op would have applicability in the GPU programming model but I'm not sure about the semantics you are looking for. Should we schedule a 1:1 to discuss this further?

Reply:
Yes, this is also related to discussions around scalable vectors: https://llvm.discourse.group/t/rfc-add-built-in-support-for-scalable-vector-types/4484/13
Let's plan a meeting with @ThomasRaoux too on PST time and see how we can split things up here.